### PR TITLE
Update CI

### PR DESCRIPTION
**Pull request by mileslucas**
- remove old CI
- add GHA
- update README badges and links
- update links in docs

This should only require adding the documenter keys to [secrets](https://github.com/juliaastro/AstroLib.jl/settings/secrets) to get the docs working

### CODE REVIEW

**Comment by mileslucas** [comment]
hm I removed `appveyor.yml` but it still seems to be building. That might need to be removed from the appveyor settings.

Or I can remove the windows build form the matrix and leave the appveyor how it is.

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        arch: 
+          - x64
+    steps:
+      - uses: actions/checkout@v2
+      - uses: julia-actions/setup-julia@v1
+        with:
+          version: ${{ matrix.version }}
+          arch: ${{ matrix.arch }}
+      - name: Cache artifacts
+        uses: actions/cache@v1
```
Comment (giordano) [inline .github/workflows/CI.yml:33]:
```suggestion
        uses: actions/cache@v2
```

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      - uses: julia-actions/julia-processcoverage@v1
+      - uses: codecov/codecov-action@v1
+        with:
+          file: ./lcov.info
+          flags: unittests
+      - uses: coverallsapp/github-action@master
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          path-to-lcov: ./lcov.info
+          flag-name: Unit
```
Comment (giordano) [inline .github/workflows/CI.yml:53]: Why not
```suggestion
      - uses: julia-actions/julia-uploadcodecov@v0.1
        continue-on-error: true
```
?

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  docs:
+    name: Documentation
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: julia-actions/setup-julia@v1
+        with:
+          version: "1"
```
Comment (giordano) [inline .github/workflows/CI.yml:61]: Better to cache artifacts also here (but I wish this was much simpler...)
```suggestion
          version: "1"
      - name: Cache artifacts
        uses: actions/cache@v2
        env:
          cache-name: cache-artifacts
        with:
          path: ~/.julia/artifacts
          key: ${{ runner.os }}-test-${{ env.cache-name }}-${{ hashFiles('**/Project.toml') }}
          restore-keys: |
            ${{ runner.os }}-test-${{ env.cache-name }}-
            ${{ runner.os }}-test-
            ${{ runner.os }}-

```

File: .github/workflows/CompatHelper.yml [hunk 4]
```diff
@@ -1,19 +1,15 @@
 name: CompatHelper
-
 on:
   schedule:
-    - cron: '00 23 * * *'
-
+    - cron: 0 23 * * *
 jobs:
-  CompatHelper:
+  build:
```
Comment (giordano) [inline .github/workflows/CompatHelper.yml:6]: Uh? :slightly_smiling_face: 
```suggestion
  CompatHelper:
```

File: .github/workflows/CI.yml (hunk 2 above)
Comment (mileslucas) [reply .github/workflows/CI.yml:53]: I was mostly just copying the config from other repos 😛

File: .github/workflows/CompatHelper.yml (hunk 4 above)
Comment (mileslucas) [reply .github/workflows/CompatHelper.yml:6]: just copying from other projects- this still works I think

**Comment by giordano** [comment]
Looking good, thanks!

**Pull request merged**